airflow/plugins/operators/scrape_ntd_xlsx.py

```diff
@@ -128,43 +134,57 @@ def __init__(
-                df.to_json(orient="records", lines=True).encode()
-            )
-
-            tab_name = ""
-
-            tab_name = make_name_bq_safe(key)
-
-            # Save clean gzipped jsonl files to the clean bucket
-            self.clean_excel_extract = CleanExtract(
-                year=self.year,
-                product=self.product + "/" + self.year + "/" + tab_name,
-                filename=f"{self.year}__{self.product}__{tab_name}.jsonl.gz",
-            )
-
-            self.clean_excel_extract.save_content(
-                fs=get_fs(), content=self.clean_gzipped_content
-            )
+        try:
+            download_url = self.raw_excel_extract.file_url
+
+            key = (self.product, self.year)
+            if key in xcom_keys:
+                download_url = pull_url_from_xcom(key=xcom_keys[key], context=context)
+
+            logging.info(f"reading {self.product} url as {download_url}")
+
+            excel_content = self.raw_excel_extract.fetch_from_ntd_xlsx(download_url)
+            if excel_content is None:
+                return None
+
+            # Save raw content
+            self.raw_excel_extract.save_content(fs=get_fs(), content=excel_content)
+
+            # Process Excel file
+            excel_data = BytesIO(excel_content)
+            try:
+                df_dict = pd.read_excel(excel_data, sheet_name=None, engine="openpyxl")
+            except Exception as e:
+                logging.error(f"Error reading Excel file: {e}")
+                raise AirflowException(f"Failed to read Excel file: {e}")
+
+            # Process each sheet
+            for key, df in df_dict.items():
+                try:
+                    df = df.rename(make_name_bq_safe, axis="columns")
+                    logging.info(f"read {df.shape[0]} rows and {df.shape[1]} columns")
+
+                    self.clean_gzipped_content = gzip.compress(
+                        df.to_json(orient="records", lines=True).encode()
+                    )
+
+                    tab_name = make_name_bq_safe(key)
+
+                    # Save clean gzipped jsonl files to the clean bucket
+                    self.clean_excel_extract = CleanExtract(
+                        year=self.year,
+                        product=self.product + "/" + self.year + "/" + tab_name,
+                        filename=f"{self.year}__{self.product}__{tab_name}.jsonl.gz",
+                    )
+
+                    self.clean_excel_extract.save_content(
+                        fs=get_fs(), content=self.clean_gzipped_content
+                    )
+
+                except Exception as e:
+                    logging.error(f"Error processing sheet {key}: {e}")
+                    raise AirflowException(f"Failed to process Excel sheet {key}: {e}")
+
+        except Exception as e:
+            logging.error(f"Error in XLSX operator execution: {e}")
+            raise AirflowException(f"XLSX operator execution failed: {e}")
```

Comment:
For the nested exceptions here, this last one is handling just lines 138 to 154? Maybe could be moved up?